### PR TITLE
Initialize

### LINKED ISSUE #41: Implement api call to robot status
**Describe the new feature you would like to see**
Get the robot status from the Energy Robotics Api

### PR DESCRIPTION
**Pull request by MortFred**
Closes #41 and #27

### CODE REVIEW

File: src/isar_exr/api/energy_robotics_api.py [hunk 1]
```diff
@@ -66,3 +68,42 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+        params: dict = {"robot_id": exr_robot_id}
+        try:
+            result: Dict[str, Any] = self.client.query(query_string, params)
+        except Exception as e:
+            raise RobotException(e)
+
+        while not self.get_robot_awake_status(exr_robot_id):
+            sleep(1)
+
+    def get_robot_awake_status(self, exr_robot_id: str) -> bool:
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:91]:
```suggestion
    def is_robot_awake(self, exr_robot_id: str) -> bool:
```

File: src/isar_exr/api/energy_robotics_api.py [hunk 2]
```diff
@@ -66,3 +68,42 @@ def pause_current_mission(self, exr_robot_id: str) -> None:
...
+            }
+        """
+        params: dict = {"robot_id": exr_robot_id}
+        try:
+            result: Dict[str, Any] = self.client.query(query_string, params)
+        except Exception as e:
+            raise RobotException(e)
+
+        while not self.get_robot_awake_status(exr_robot_id):
+            sleep(1)
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:89]: We should probably have a max wait time limit in the config.
After the max wakeup time, we fail assuming the robot cannot wake up. Maybe it is then dead

**Review by GodVenn — CHANGES_REQUESTED** [review]

File: src/isar_exr/config/settings.py [hunk 3]
```diff
@@ -11,6 +11,8 @@ class Settings(BaseSettings):
     # URL for Ex-Robotics API
     ROBOT_API_URL: str = Field(default="https://developer.energy-robotics.com/graphql/")
 
+    MAX_WAKE_TIME_S: int = 120
```
Comment (GodVenn) [inline src/isar_exr/config/settings.py:14]:
```suggestion
    # The maximum time (in seconds) we wait for the robot to wake up after sending the wake up command
    MAX_TIME_FOR_WAKEUP: int = 120
```

**Review by GodVenn — COMMENTED** [review]
Suggesting a rename and adding a comment explaining the variable

**Review by GodVenn — CHANGES_REQUESTED** [review]
Can we add tests for `wake_up_robot` and `is_robot_awake` ?

File: src/isar_exr/api/energy_robotics_api.py [hunk 4]
```diff
@@ -205,19 +207,22 @@ def wake_up_robot(self, exr_robot_id: str) -> None:
...
-            if (datetime.now() - startTime).seconds > settings.MAX_WAKE_TIME_S:
+        awake = self.is_robot_awake(exr_robot_id)
+        while not awake:
+            time_passed_since_function_call = (
+                datetime.today() - startTime
+            ).total_seconds()
+            if time_passed_since_function_call > settings.MAX_TIME_FOR_WAKEUP:
                 raise RobotException(
-                    f"Not able to wake up robot after '{settings.MAX_WAKE_TIME_S}' seconds."
+                    f"Not able to wake up robot after '{settings.MAX_TIME_FOR_WAKEUP}' seconds."
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:217]: I think this won't work? `awake` is never updated inside the while loop so the while loop will always time out
The previous method of while not self.is_robot_awake works though

File: src/isar_exr/api/energy_robotics_api.py [hunk 5]
```diff
@@ -188,12 +188,14 @@ def create_waypoint_task_definition(
             raise RobotException(e)
 
         return response_dict["createWaypointTaskDefinition"]["id"]
-    
-    def wake_up_robot(self, exr_robot_id: str) -> None:
+
+    def wake_up_robot(
+        self, exr_robot_id: str, startTime: datetime = datetime.today()
```
Comment (GodVenn) [inline src/isar_exr/api/energy_robotics_api.py:193]: Not sure it makes sense to pass startTime as an input to this function?
What does it mean to the functionality of "wake_up_robot" ?

**Review by GodVenn — COMMENTED** [review]
The commits don't really reflect what they change, is it possible to take parts of the last commit "add tests" and instead squash them into their appropriate commits?

File: src/isar_exr/api/energy_robotics_api.py (hunk 5 above)
Comment (MortFred) [reply src/isar_exr/api/energy_robotics_api.py:193]: This is done to make the function testable. Or else, we have to wait the full MAX_TIME_FOR_WAKEUP to test whether or not the function throws an exeption at timeout

Comment (GodVenn) [reply src/isar_exr/api/energy_robotics_api.py:193]: Couldn't we instead change the settings for the test?

**Pull request merged**